### PR TITLE
AKS on Azure Stack HCI GitOps Basic

### PR DESCRIPTION
**Pull request by lanicolas**
Automation for GitOps configuration for AKS on Azure Stack HCI

### CODE REVIEW

File: azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+az account set --subscription $subscriptionId
+
+#Get AKS on Azure Stack HCI cluster credentials
+Get-AksHciCredential -Name $ClusterName 
+
+# Create a namespace for your ingress resources
+kubectl create namespace hello-arc
+
+# Helm Install 
+
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1:25]:
```suggestion

```

Comment (likamrat) [reply azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1:25]: I am missing the automation for the k8s cli extensions (connectedk8s and k8s-configuration)

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+* [Install or update Azure PowerShell modules](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps?view=azps-5.6.0). Use the below command to check your current installed version.
+
+  ```PowerShell
+  Get-InstalledModule -Name Az -AllVersions | select Name,Version
+  ```
+
+* Create Azure service principal (SP)
+
+    To be able to complete the scenario and its related automation, Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account run the below command.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:55]: Unlike other scenarios, here, you need to be explicit about the use of PowerShell

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+* User will kick the GitOps flow by making changes to the ["Hello Arc"](https://github.com/likamrat/hello_arc) application, this will be considered a new version by the Flux Operator.
+
+* The GitOps flow will pull the new version of the application and deploy new pods, upon committing the changes.
+
+* User will show rolling upgrade of the "Hello Arc" application as the GitOps flow is completed successfully.
+  
+## Azure Arc Kubernetes GitOps Configuration
+
+* Edit the environment variables in the [*az_k8sconfig_aks_hci*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1) shell script to match your parameters and run it using the ```.\az_k8sconfig_aks_hci.ps1``` command.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:137]:
```suggestion
* Edit the environment variables in the [*az_k8sconfig_aks_hci*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1) PowerShell script to match your parameters and run it using the ```.\az_k8sconfig_aks_hci.ps1``` command.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+    ![New GitOps configurations](./04.png)
+
+    ![New GitOps configurations](./05.png)
+
+    ![New GitOps configurations](./06.png)
+
+## The "Hello Arc" Application & Components
+
+* Before kicking the GitOps flow, let's verify and zoom-in to the Kubernetes resources deployed by running few *kubectl* commands.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:153]:
```suggestion
* Before kicking the GitOps flow, let's review the Kubernetes resources deployed by running a few *kubectl* commands.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    1. The Flux operator holds the "desired state" of the "Hello Arc" application, this are the configuration we deployed against the Azure Arc connected cluster. The operator "pulls" the state of the ["Hello Arc"](https://github.com/likamrat/hello_arc) application repository.
+
+    2. Changing the application which is consider to be a new version of it, will trigger the Flux operator to kick-in the GitOps flow.
+
+    3. A new Kubernetes pod with the new version of the application will be deployed on the cluster. Once the new pods is successfully deployed, the old one will be terminated (rolling upgrade).
+
+* To show the above flow, open 2 (ideally 3) side-by-side browser windows:
+
+  * PowerShell open running the ```kubectl get pods -n hello-arc -w``` command
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:178]:
```suggestion
  * Run the ```kubectl get pods -n hello-arc -w``` command
```

**Review by likamrat — COMMENTED** [review]
- automate the k8s cli extensions
- why removing "docs/azure_arc_jumpstart/azure_arc_k8s/day2/_index.md" ?
- Small syntax updates

File: azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1 (hunk 1 above)
Comment (lanicolas) [reply azure_arc_k8s_jumpstart/aks_stack_hci/gitops/basic/az_k8sconfig_aks_hci.ps1:25]: I cannot do the install from the PowerShell script I will include in the prerequisites

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md (hunk 2 above)
Comment (lanicolas) [reply docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:55]: Not sure exactly what I should highlight here?

Comment (lanicolas) [reply docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:57]: added the change

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+* [Install or update Azure CLI to version 2.15.0 and above](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.
+
+  ```shell
+  az --version
+  ```
+
+* Create Azure service principal (SP)
+
+    To be able to complete the scenario and its related automation, Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account run the below command.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/day2/aks_stack_hci/aks_hci_gitops_basic/_index.md:57]:
```suggestion
    To be able to complete the scenario and its related automation, an Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account run the below PowerShell command.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/aks_stack_hci/aks_hci_powershell/_index.md [hunk 7]
```diff
@@ -26,7 +26,7 @@ This guide will not provide instructions on how to deploy and set up Azure Stack
   
 * Create Azure service principal (SP)
 
-    To be able to complete the scenario and its related automation, Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account using PowerShell run the below command (this can also be done in [Azure Cloud Shell](https://shell.azure.com/)).
+    To be able to complete the scenario and its related automation, Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account using PowerShell run the below command. You will need to run the script from a PowerShell window that has access to your AKS on Azure Stack HCI environment.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/aks_stack_hci/aks_hci_powershell/_index.md:29]:
```suggestion
    To be able to complete the scenario and its related automation, an Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account using PowerShell and run the below command. To do this, you will need to run the script from a PowerShell session that has access to your AKS on the Azure Stack HCI environment. 
```

**Review by likamrat — APPROVED** [review]
Great work. LGTM

**Pull request merged**